### PR TITLE
build: migrate to monorepo (for vue-jest@5)

### PR DESCRIPTION
**Pull request by nogic1008**
⚠️ **DO NOT merge this PR with "Squash & Merge" method!!** ⚠️
It breaks [package.json](https://github.com/vuejs/vue-jest/blob/next/package.json) histories.

This PR is a part of #355.
Similar to #360, but targeted at `next`(vue-jest@5) branch.

## Changes
- setup monorepo (use [yarn workspace](https://classic.yarnpkg.com/en/docs/workspaces/))
- move `e2e/__projects__/*` projects to `e2e/3.x/*`
- move `package.json` and `lib/` to `packages/vue3-jest/`
- change package to `vue3-jest@26.0.0-alpha.10`
- fix e2e test for monorepo

### CODE REVIEW

File: packages/vue3-jest/package.json [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{
+  "name": "vue3-jest",
+  "version": "26.0.0-alpha.10",
```
Comment (lmiller1990) [inline packages/vue3-jest/package.json:3]: How do you think we should handle this? Should this match a specific Jest version (maybe latest 26.x minor version?)

File: packages/vue3-jest/package.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    "jest": "^26.0.0",
+    "semantic-release": "^15.13.2",
+    "ts-jest": "^26.4.4",
+    "typescript": "^4.1.2",
+    "vue": "^3.0.3"
+  },
+  "peerDependencies": {
+    "@babel/core": "7.x",
+    "babel-jest": ">= 24 < 27",
+    "jest": ">= 24 < 27 ",
```
Comment (lmiller1990) [inline packages/vue3-jest/package.json:41]: I think we should only match Jest 26 now.

**Review by lmiller1990 — COMMENTED** [review]
I left some comments.

I have a question that we did not think about. We will need to support multiple versions of Jest at the same time for a while - in this case, we need to maintain v26 and v27 support.

Should we have two packages? `vue3-jest26` and `vue3-jest27`? What do you think is the best way to handle this?

**Comment by nogic1008** [comment]
@lmiller1990 
I think it's appropriate to separate the branches depending on supported Jest version.

- `master` branch
  - `vue2-jest@27`
  - `vue3-jest@27`
- `26.x` branch
  - `vue2-jest@26`
  - `vue3-jest@26`

File: packages/vue3-jest/package.json (hunk 2 above)
Comment (nogic1008) [reply packages/vue3-jest/package.json]: Changed to `26.x`.

**Comment by lmiller1990** [comment]
@nogic1008 I see. I wonder if there will be any complexity in having both Vue 2 and Vue 3 in the same monorepo? Are we able to do this (I guess we need to do `nohoist`?)

Is there anything else you'd like to do before merging this? Otherwise, I'll merge and do a release to make sure everything is 👌 . Then we can work on getting Jest 27 support merged.

**Comment by nogic1008** [comment]
I believe that the complexity of including different Vue version in `node_modules` is less than the complexity of having to separate the same issues on different branches. (ex. security update by dependabot)
I copied /packages and /e2e from `next` branch to `master` branch locally, but it seems that `nohoist` settings are not needed at this time.

We need to merge `next` branch into `master` branch to complete the monorepo.
README also needs to be added and updated. (in root and each packages)

**Comment by lmiller1990** [comment]
Ok, let's try your idea. 

I think we can update the README later, next goal should be Jest 27.

**Pull request merged**